src/cluster.c

```diff
@@ -778,6 +778,33 @@ int isValidAuxString(char *s, unsigned int length) {
     return 1;
 }
 
+int parseAndValidateConnectionPorts(client *c, long long *port, long long *cport) {
+    if (getLongLongFromObject(c->argv[3], port) != C_OK) {
+        addReplyErrorFormat(c, "Invalid base port specified: %s", (char *)c->argv[3]->ptr);
```

Comment:
This function has a strong coupling with the CLUSTER MEET command. The `c->argv[3]` and `c->argv[4]` are hard-coded to match exactly the args of that command. It is actually a helper function that's only usable for custer meet.

I think it's better just keep this code directly in clusterCommandSpecial where cluster meet is handled. If we keep it there, it is clear from the surrounding context what argv[3] and argv[4] are.

We can do some other simplification of the validation, like function to parse and validate a port, such as `int getPortFromObject(robj *arg, int *port)` that just parses the number and does a port range check.